sigstore/_cli.py

```diff
@@ -416,24 +436,31 @@ def _verify(args: argparse.Namespace) -> None:
         if not file.is_file():
             args._parser.error(f"Input must be a file: {file}")
 
-        sig, cert = args.signature, args.certificate
+        sig, cert, bundle = args.signature, args.certificate, args.rekor_bundle
         if sig is None:
             sig = file.parent / f"{file.name}.sig"
         if cert is None:
             cert = file.parent / f"{file.name}.crt"
+        if bundle is None:
+            bundle = file.parent / f"{file.name}.bundle"
 
         missing = []
         if not sig.is_file():
             missing.append(str(sig))
         if not cert.is_file():
             missing.append(str(cert))
+        if not bundle.is_file() and args.rekor_offline:
+            # NOTE: We only produce errors on missing bundle files
+            # if the user has explicitly requested offline-only verification.
+            # Otherwise, we fall back on online verification.
+            missing.append(str(bundle))
```

Comment:
This seems like it'd be somewhat surprising behavior -- if the bundle file isn't valid, we'll silently attempt online verification and then potentially fail if that's unavailable? Can't the presence of `--rekor-bundle` imply that the user wanted offline verification of _only_ that bundle?

Reply:
+1, this seems like a case where we want to alert the user that their argument was invalid.

We used to fallback on online verification when the bundle produced a failure in cosign, and then we decided it was not correct. Granted, here it's not even getting to that point.

Reply:
Yeah, I agree that this is surprising as-is: we should treat `--rekor-bundle` as implying `--rekor-offline`. I'll tweak to that.

Reply:
https://github.com/sigstore/sigstore-python/pull/247/commits/34a1e4a4291c952c4bc6640aaa5dd1f754386cf2 accomplishes this!